### PR TITLE
feat: use golangci-lint-action

### LINKED ISSUE #81: Use golangci GitHub Action
We can replace [this step](https://github.com/aquasecurity/trivy-db/blob/086d97d517369361ca78e5f063f51854b0cb2b58/.github/workflows/go.yml#L18) with the [`golangci-lint-action`](https://github.com/golangci/golangci-lint-action).

I shall submit a PR if this a valid issue.

### PR DESCRIPTION
**Pull request by sudiptog81**
* utilise golangci-lint-action instead of installing and running it
* closes #81

### CODE REVIEW

**Comment by sudiptog81** [comment]
These error logs might be of interest.

```logs
##[error]Error return value of `(*go.etcd.io/bbolt.Bucket).ForEach` is not checked (errcheck)
##[error]Error return value of `(*go.etcd.io/bbolt.Bucket).ForEach` is not checked (errcheck)
##[error]S1031: unnecessary nil check around range (gosimple)
##[error]type `args` is unused (unused)
```

**Comment by knqyf263** [comment]
As for `errcheck`, you can add `nolint`. We probably should fix errors from `gosimple` and `unused`.

**Comment by sudiptog81** [comment]
The following errors might give an insight. Ran `golangci-lint run --fix -D errcheck` to get this.

``` bash
pkg/vulnsrc/python/python_test.go:78:5: S1031: unnecessary nil check around range (gosimple)
                                if adv != nil {
                                ^
pkg/vulnsrc/python/python_test.go:17:7: type `args` is unused (unused)
```

**Comment by knqyf263** [comment]
You can remove the if statement.

**Comment by sudiptog81** [comment]
The test is passing now.

**Comment by knqyf263** [comment]
Do you have any reason to disable `unused`?

**Comment by sudiptog81** [comment]
The tests were failing due to https://github.com/aquasecurity/trivy-db/blob/master/pkg/vulnsrc/python/python_test.go#L17-L19 which may be necessary but causes the `unused` linter rule to fail.

**Comment by knqyf263** [comment]
It looks like unnecessary. You can remove it.

**Comment by sudiptog81** [comment]
I had gone ahead and removed those lines. The test is failing a https://github.com/aquasecurity/trivy-db/runs/1271265059#step:4:70.

File: pkg/vulnsrc/python/python_test.go [hunk 1]
```diff
@@ -14,9 +14,6 @@ import (
 )
 
 func TestVulnSrc_Update(t *testing.T) {
-	type args struct {
-		dir string
-	}
 	testCases := []struct {
 		name string
 		args struct {
```
Comment (knqyf263) [inline pkg/vulnsrc/python/python_test.go:19]: What if defining this struct outside like before? And use it as follows.

```
	type args struct {	
		dir string	
	}
	...
	testCases := []struct {
	...
		args args
```

Comment (sudiptog81) [reply pkg/vulnsrc/python/python_test.go]: The tests are now passing after making the suggested changes.

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
Thanks.

**Pull request merged**